script/validate_json.rb

```diff
@@ -1,53 +1,79 @@
 #!/usr/bin/env ruby
 
 # Validates JSON files in the _data directory
-# Exits 0 on success
-# Exits 1 upon JSON parsing errors
-# Exits 2 if a file's keys are not in alphanumeric order
-# Exits 3 if any sites.json entries are missing the required 'url' key
 
 require 'json'
 
+module ExitCodes
+    SUCCESS = 0
+    PARSE_FAILED = 1           # JSON parse errors
+    UNSORTED = 2               # data keys are not in alphanumeric order
+    MISSING_URL = 3            # Entry missing the required 'url' field
+    MISSING_DIFFICULTY = 4     # Entry missing the required 'difficulty' field
+    MISSING_DOMAINS = 5        # Entry missing the required 'domains' field
+    MISSING_NAME = 6           # Entry missing the required 'name' field
+    UNEXPECTED_DIFFICULTY = 7  # Unexpected value for 'difficulty' field
+end
+
+SupportedDifficulties = ["easy","medium","hard","impossible"]
```

Comment:
Add spaces after commas, and also within brackets since this is a one-liner.
```suggestion
SupportedDifficulties = [ "easy", "medium", "hard", "impossible" ]
```

Reply:
I added after commas, but not within brackets because the others in the file didn't seem to be like this.